[wjayee] iP

wjayee opened the pull request:
# Duke: Toothless
> It’s not a bug; it’s an undocumented feature. ― Anonymous

What's so good about toothless? It's
- Smart
- ~~Hard~~ _EASY_ to use

Where can I find this?
1. Download it from [here](https://github.com/nus-cs2103-AY2324S1/ip/pull/209)
2. Run the file
3. Give commands to it
4. Enjoy the features :slightly_smiling_face: 

And the best thing is that its **FREE TO USE!**

Supported features include:
- [ ] Keep track of different types of tasks
- [ ] Edit list of tasks

If you want to built your own version of toothless, here's the `main` method 
```
public class Duke {
    public static void main(String[] args) {
        String logo = " ____        _        \n"
                + "|  _ \\ _   _| | _____ \n"
                + "| | | | | | | |/ / _ \\\n"
                + "| |_| | |_| |   <  __/\n"
                + "|____/ \\__,_|_|\\_\\___|\n";
        System.out.println("Hello from\n" + logo);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

wjayee committed:
Level-0

wjayee committed:
Level-1

wjayee committed:
Level-2

wjayee committed:
Level-3

wjayee committed:
Level-4

wjayee committed:
A-TextUiTesting

wjayee committed:
Level-5

wjayee committed:
Level-6

wjayee committed:
A-Enums

wjayee committed:
branch-Level-7

wjayee committed:
Merge branch 'branch-Level-7'

wjayee committed:
Merge branch 'master' of https://github.com/wjayee/ip

wjayee committed:
Add Level-8

wjayee committed:
Merge branch 'branch-Level-8'

wjayee committed:
Add A-MoreOOP

wjayee committed:
Merge branch 'branch-A-MoreOOP'

wjayee committed:
Add A-Packages

wjayee committed:
Merge remote-tracking branch 'origin/add-gradle-support'

wjayee committed:
Add A-Gradle

wjayee committed:
Add A-JUnit

wjayee committed:
Add A-Jar

wjayee committed:
Add A-CodingStandard

wjayee committed:
Add A-JavaDoc

wjayee committed:
Add Level-9

wjayee committed:
Merge branch 'branch-A-JavaDoc'

wjayee committed:
Merge branch 'branch-A-CodingStandard'

wjayee committed:
Resolve conflicts

wjayee committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/tasks/TaskList.java
#	src/main/java/duke/utils/Parser.java

wjayee committed:
Add CheckStyle

wjayee committed:
Merge branch 'branch-A-CheckStyle'

Heran9 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,114 @@
+            ui.separator();
+        }
+        ui.farewell();
+    }
```
Maybe can add some javadocs here, thus it would be easier for reader to understand.

Heran9 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,114 @@
+                case TODO:
+                case DEADLINE:
+                case EVENT:
+                    Task t = TaskList.createTask(nextInput, command, 0);
```
Consider using more descriptive variable and method names where applicable. For example, instead of t, use task.

Heran9 commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,77 @@
+     */
+    public List<String> readFile() {
+        List<String> output = new ArrayList<>();
+        File f = new File(path);
```
Better to use file instead of f

Heran9 commented on src/main/java/duke/tasks/TaskList.java:
```diff
@@ -0,0 +1,196 @@
+     * @throws DukeException If title or date does not exist, or if input is invalid.
+     * @throws DateTimeParseException If date exists but is in the wrong format.
+     */
+    public static Task createTask(String input, Commands command, int isDone)
```
Nice line feed to avoid a too long sentence.

Heran9 commented on src/main/java/duke/tasks/TaskList.java:
```diff
@@ -0,0 +1,196 @@
+
+        } catch (IndexOutOfBoundsException ioob) {
+            throw new MissingTaskException("Missing Task");
+        } catch (Exception e) {
+            throw new InvalidInputException("Invalid input");
+        }
+    }
+
+    /**
+     * Returns tasks that matches the keyword provided by the user.
+     * @param input The user input.
+     * @return The tasks that matched the keyword.
+     * @throws DukeException If user did not provide a keyword or if the input was invalid.
+     */
+    public List<String> findTask(String input) throws DukeException {
+        try {
+            String keyword = Parser.obtainTitle(input, Commands.FIND);
+
+            List<String> results = new ArrayList<>();
+            for (int i = 0; i < tasks.size(); i++) {
+                Task t = tasks.get(i);
+                if (t.getTask().contains(keyword)) {
+                    results.add(t.getTask());
+                }
+            }
+
+            return results;
+        } catch (DukeException e) {
+            throw e;
+        }
+    }
+
+    /**
+     * Adds a task to the list of tasks.
+     * @param task The task to be added.
+     */
+    public void addTask(Task task) {
+        tasks.add(task);
+    }
+
```
Overall really good Javadocs with specificed details and clear parameter names.

Heran9 reviewed (commented):
Overall very good in coding standard.

ktzy0305 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,114 @@
+     */
+    public void run() {
+
+        boolean end = false;
```
Perhaps add a prefix to make this sound more boolean like. Eg: `isEnd`

ktzy0305 commented on src/main/java/duke/exceptions/InvalidInputException.java:
```diff
@@ -0,0 +1,23 @@
+     * Returns the description of the exception.
+     * @return Description of exception.
+     */
+    @Override
+    public String getMessage() {
+        return "I don't understand what you just said :(";
+    }
```
Consider avoiding the use of "magic literals" in general.

I would suggest adding a named constant as a member of the InvalidInputException class:
```
public class InvalidInputException extends DukeException {
    private static final String MESSAGE = "I don't understand what you just said :("

    // Other code and JavaDocs

    @Override
    public String getMessage() {
        return MESSAGE;
    }
}
```

ktzy0305 commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,77 @@
+     * @param fw File to be written in.
+     * @param tasks Tasks to be written.
+     * @throws IOException If an I/O error occurs.
+     */
```
Do add an empty line between the method description and the parameters section when writing JavaDocs for methods

ktzy0305 commented on src/main/java/duke/utils/Parser.java:
```diff
@@ -0,0 +1,98 @@
+
+            } else {
+                throw new InvalidInputException("Invalid input");
+            }
```
Great job on avoiding deep nesting in your code! 👍

ktzy0305 commented on src/main/java/duke/tasks/TaskList.java:
```diff
@@ -0,0 +1,196 @@
+     * @return A list of description to be stored locally if typeOfDes = 0 else
+     *     a list of description to be read by the user.
+     */
+    public List<String> getTasksDes(int typeOfDes) {
```
Perhaps using the full method name `getTaskDescription(...)` would enhance readability.

ktzy0305 commented on src/main/java/duke/utils/Parser.java:
```diff
@@ -0,0 +1,98 @@
+            throw new InvalidInputException("Invalid input");
+        }
+    }
+}
```
Overall good job on making the variable and method names descriptive and detailed! 👍

ktzy0305 reviewed (commented):
The code quality is generally great!

wjayee committed:
Add Level-10 v1

wjayee committed:
Add Level-10 v2

wjayee committed:
Add Level-10 v3

wjayee committed:
Merge branch 'branch-Level-10'

wjayee committed:
Add assert to Duke.java

Assert not implemented in Duke.java

Assert will help to check the assumptions made in Duke

Check that the response to the user's input is not empty.

Ensures that users are given a description of the error
if their command cannot be recognized

wjayee committed:
Add assert to TaskList.java.

Assert not implemented in TaskList.java.

Assert will help to check the assumptions made in methods in
TaskList.java.

Check that the commands and the String inputs passed into
some methods are valid.

Ensures that methods are called correctly in other classes.

wjayee committed:
Change code quality

Certain code in some files are not of good quality

Minor changes are made to improve the code quality of some methods

It improves its readability and removes code that are error prone

wjayee committed:
Merge pull request #1 from wjayee/branch-A-CodeQuality

Change code quality

wjayee committed:
Merge branch 'master' into branch-A-Assertions

wjayee committed:
Merge pull request #2 from wjayee/branch-A-Assertions

Branch a assertions

johnnythesnake12 commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -32,10 +32,12 @@ public List<String> readFile() {
         try {
             f.createNewFile();
             Scanner s = new Scanner(f);
```
Generally good code quality, but maybe can consider using File file, Scanner scanner and even below Filewriter filewriter to make it even easier for others to follow

wjayee committed:
Add new feature to manage notes

Previous versions did not allow notes to be stored.

New classes like Note and NoteList are added to store notes alongside
tasks.

New classes are added for abstraction, rather than editing the current
Task class.

wjayee committed:
Merge branch 'branch-D-Notes'

wjayee committed:
Add User Guide

There is no user guide currently.

User guide is added to the readme for easy viewing

wjayee committed:
Merge branch 'branch-A-UserGuide'

wjayee committed:
Add UserGuide v2

wjayee committed:
Remove the bug where the note list is not regenerated after closing
the program

Opening the program doesn't reload the previously saved notes
correctly.

Change the format that notes are saved in the local file.

wjayee committed:
no message

wjayee committed:
Merge branch 'master' of https://github.com/wjayee/ip
